python/mspasspy/db/database.py

```diff
@@ -291,57 +358,62 @@ def _save_elog(self, data, collection='error_logs'):
         elog is the error log object to be saved.
         Return:  List of ObjectID of inserted
         """
-        if 'elog_ids' not in data:
-            data['elog_ids'] = []
+        if not collection:
+            collection = self.database_schema.default_name('elog')
+
+        if not elog_id:
+            elog_id = []
+
         oid = None
-        if '_id' in data:
-            oid = data['_id']
-        elog = data.elog
+        if '_id' in mspass_object:
+            oid = mspass_object['_id']
+
+        if isinstance(mspass_object, Seismogram):
+            wf_id_name = 'wf_Seismogram_id'
+        elif isinstance(mspass_object, TimeSeries):
+            wf_id_name = 'wf_TimeSeries_id'
+        else:
+            raise TypeError("only TimeSeries and Seismogram are supported")
```

Comment:
This exception also needs a clean-up. This should be handled in the methods that calls `_save_elog`.